zerver/migrations/0425_userpresence_fill_data.py

```diff
@@ -0,0 +1,62 @@
+from django.db import connection, migrations
+from django.db.backends.postgresql.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+
+
+def fill_new_columns(apps: StateApps, schema_editor: BaseDatabaseSchemaEditor) -> None:
+    UserPresence = apps.get_model("zerver", "UserPresence")
+
+    # In theory, we'd like to preserve the distinction between the
+    # IDLE and ACTIVE statuses in legacy data.  However, there is no
+    # correct way to do so; the previous data structure only stored
+    # the current IDLE/ACTIVE status of the last update for each
+    # (user, client) pair. There's no way to know whether the last
+    # time the user had the other status with that client was minutes
+    # or months beforehand.
+    #
+    # So the only sane thing we can do with this migration is to treat
+    # the last presence update as having been a PRESENCE_ACTIVE_STATUS
+    # event. This will result in some currently-idle users being
+    # incorrectly recorded as having been active at the last moment
+    # that they were idle before this migration.  This error is
+    # unlikely to be significant in practice, and in any case is an
+    # unavoidable flaw caused by the legacy previous data model.
+    with connection.cursor() as cursor:
+        cursor.execute(
+            "SELECT realm_id, user_profile_id, MAX(timestamp) FROM zerver_userpresenceold WHERE status IN (1, 2) GROUP BY realm_id, user_profile_id"
+        )
+        latest_presence_per_user = cursor.fetchall()
```

Comment:
This is doable directly in Django if you prefer:
```python3
UserPresenceOld.objects.values('realm_id','user_profile_id').annotate(latest=Max('timestamp')).filter(status__in=[1,2])
```

Reply:
I can swap it to the ORM query, though I went with SQL on purpose because I somehow don't find it very understandable into what `GROUP BY` SQL exactly Django will translate `.annotate` queries - and with raw SQL we can know exactly what it's going to do :neutral_face:

Reply:
Yeah it's nice to use the ORM, but I also find Django's annotate syntax a little hard to read, so don't have a strong opinion on this one way or another.